pkg/crc/preflight/preflight_darwin.go

```diff
@@ -70,13 +71,23 @@ var trayLaunchdCleanupChecks = []Check{
 	{
 		configKeySuffix:  "check-old-autostart",
 		checkDescription: "Checking if old launchd config for tray autostart exists",
-		check:            func() error { return fmt.Errorf("force trigger cleanup to remove old launchd config for tray") },
-		fixDescription:   "Removing old launchd config for tray autostart",
-		fix: func() error {
-			_ = os.Remove(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.tray.plist"))
-			_ = os.Remove(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.daemon.plist"))
+		check: func() error {
+			if crcos.FileExists(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.tray.plist")) {
+				return fmt.Errorf("force trigger cleanup to remove old launchd config for tray")
+			}
 			return nil
 		},
+		fixDescription: "Removing old launchd config for tray autostart",
+		fix: func() error {
+			var err error
+			if crcos.FileExists(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.tray.plist")) {
+				err = os.Remove(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.tray.plist"))
+			}
+			if crcos.FileExists(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.daemon.plist")) {
+				err = os.Remove(filepath.Join(constants.GetHomeDir(), "Library", "LaunchAgents", "crc.daemon.plist"))
+			}
+			return err
```

Comment:
I can't follow what is expected from 'err' here?
If removal of `crc.tray.plist` fails but removal of `crc.daemon.plist` succeeds, we ignore the error.
If removal of `crc.tray.list` succeeds but removal of `crc.daemon.plist` fails, we return the error.
Is this the intended behaviour?